aten/src/ATen/cudnn/Descriptors.h

```diff
@@ -44,29 +81,33 @@ static inline void fixSizeOneDimStride(int dim, const int *size, int *stride) {
 template <typename T, cudnnStatus_t (*dtor)(T*)>
 struct DescriptorDeleter {
   void operator()(T* x) {
-    CUDNN_CHECK(dtor(x));
+    if (x != nullptr) {
+      CUDNN_CHECK(dtor(x));
+    }
   }
 };
 
 // A generic class for wrapping cuDNN descriptor types.  All you need
 // is to give the underlying type the Descriptor_t points to (usually,
 // if it's cudnnTensorDescriptor_t it points to cudnnTensorStruct),
 // the constructor and the destructor.  Subclasses are responsible
-// for forwarding constructors and defining a set() function to actually
-// set the descriptor.
+// for defining a set() function to actually set the descriptor.
 template <typename T, cudnnStatus_t (*ctor)(T**), cudnnStatus_t (*dtor)(T*)>
 class Descriptor
 {
 public:
-  explicit Descriptor() {
-    T* raw_desc;
-    CUDNN_CHECK(ctor(&raw_desc));
-    desc_.reset(raw_desc);
-  }
-
   // TODO: Figure out why const-correctness doesn't work here
   T* desc() const { return desc_.get(); }
   T* desc() { return desc_.get(); }
+  T* mut_desc() { init(); return desc_.get(); }
```

Comment:
```
commit 4283e2aa899cb2008f9dea837a2923dfce09f07e
Author: Edward Z. Yang <ezyang@fb.com>
Date:   Fri Jan 26 09:48:30 2018 -0800

    Lazily initialize cuDNN descriptors
    
    Previously, cuDNN descriptors were eagerly allocated as soon
    as a FooDescriptor object was created.  However, in some uses
    of TensorDescriptor, this is problematic: some tensors are optional
    and cuDNN's API expects to be given a nullptr TensorDescriptor
    in this case, not an uninitialized (but allocated) descriptor.
    
    Lazily initializing the descriptors makes it less likely for
    us to use uninitialized memory and matches the usual semantics of
    unique_ptr.  It's good sense!
    
    Signed-off-by: Edward Z. Yang <ezyang@fb.com>
```

The name `mut_desc` comes from "you should use this method if you plan to mutate the descriptor". I'll comment that.

Reply:
Yeah comments in commit messages are nice, but that shouldn't mean you don't put them in the code. I never check the commit messages when reading the code, since there are often tens of commits that touch a single file.